executor/memory_test.go

```diff
@@ -114,3 +122,17 @@ func (s *testMemoryLeak) memDiff(m1, m2 uint64) uint64 {
 	}
 	return m2 - m1
 }
+
+func (s *testGlobalTracker) TestGlobalMemoryTrackerOnCleanUp(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("create table t (id int)")
+	tk.MustExec("insert t (id) values (1)")
+	tk.MustExec("insert t (id) values (2)")
+	tk.MustExec("insert t (id) values (3)")
+	tk.MustExec("update t set id = 4 where id = 1")
+	tk.MustExec("update t set id = 5 where id = 2")
+	tk.MustExec("update t set id = 6 where id = 3")
+	c.Assert(executor.GlobalMemoryUsageTracker.MaxConsumed(), Greater, int64(0))
+	c.Assert(executor.GlobalMemoryUsageTracker.BytesConsumed(), LessEqual, int64(0))
```

Comment:
During Test I found that after finishing executing the dml statement, the `BytesConsumed` of the `GlobalMemoryTracker` is note 0.

After I dig into the source code, I fount that the un-zero result caused by the following code:

https://github.com/pingcap/tidb/blob/df6898d524134703f107154ca4d5166187b9d455/session/txn.go#L547-L549

https://github.com/pingcap/tidb/blob/df6898d524134703f107154ca4d5166187b9d455/session/txn.go#L569-L571

It seems the calculating logic for the memory usage between consuming and returning is different. Not sure whether it is a bug or not. WDYT? @qw4990

Reply:
after update master branch, the test passed.